src/components/LoginForm/LoginForm.tsx

```diff
@@ -0,0 +1,108 @@
+import { useState } from "react";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import Box from "@mui/material/Box";
+import Button from "@mui/material/Button";
+import Typography from "@mui/material/Typography";
+import TextField from "@mui/material/TextField";
+import FilledInput from "@mui/material/FilledInput";
+import FormControl from "@mui/material/FormControl";
+import IconButton from "@mui/material/IconButton";
+import InputLabel from "@mui/material/InputLabel";
+import InputAdornment from "@mui/material/InputAdornment";
+
+import { CustomIcon } from "../../components";
+
+import { loginFormSchema } from "../../schemas";
+import { useAppDispatch } from "../../hooks";
+import { login } from "../../redux";
+import { ILoginRequestDTO } from "../../interfaces-submodule/interfaces/dto/auth/iadmin-login-request.interface";
+import {
+  formStyles,
+  deviderStyles,
+  inputStyles,
+  passwordIconStyles,
+  fieldWrapperStyles,
+  errorMessageStyles,
+} from "./LoginFormStyles";
+
+export const LoginForm = () => {
+  const [showPassword, setShowPassword] = useState<boolean>(false);
```

Comment:
- [ ] move all props and actions that are related to the form, to the separate hook

Reply:
@AndriiKasian I've moved all the props and actions related to the form to a separate hook as suggested.